aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -1101,9 +1102,13 @@ ReduceConfig setReduceConfig(const TensorIterator& iter){
     config.output_mult[1] = config.split_output(block_height);
   }
 
-  constexpr int min_values_per_thread = 16;
-  constexpr int max_values_per_thread = 256;
-  const int blocks_per_sm = at::cuda::getCurrentDeviceProperties()->maxThreadsPerMultiProcessor / (block_width * block_height);
+  int min_values_per_thread = 16;
+  int max_values_per_thread = 256;
+  if (config.vectorize_input) {
+    min_values_per_thread *= config.input_vec_size;
+    max_values_per_thread *= config.input_vec_size;
+  }
```

Comment:
This part I'm not 100% sure about. Is it alright to have larger limits of values per thread when we do vectorised loads, as 4 of them will be reduced sequentially by each thread after each load, or is it better to keep these max values per thread and then reorganise the logic in the next `if`?

Also, I just realised that we also have a `config.values_per_thread() >= 256` in the previous if, so if this fix of increasing the min/max values per thread is correct, we should move it up to the previous if as well and rewrite that as ` >= max_values_per_thread`.

Reply:
update: increasing `max_values_per_thread` does no good. I'll keep looking into this